core/math/camera_matrix.cpp

```diff
@@ -398,7 +398,7 @@ void CameraMatrix::invert() {
 
 		/** Product of pivots, gives determinant when finished **/
 		determinat *= pvt_val;
-		if (Math::absd(determinat) < 1e-7) {
+		if (Math::absd(determinat) < (real_t)1e-7) {
```

Comment:
This code is sus. It's calling `absd` (absolute double) on a `real_t`. It should be calling `abs` instead. Same with the two calls on line 391.

Also, this value would likely make more sense as `CMP_EPSILON` or just `1e-6`, since `1e-7` is so small of an epsilon to the point of not being very useful (`float` precision around 1 is 2^-23 or about `1.192e-7`). It's possible that `1e-7` works for this case, but it's very sus.

The same problems also exist in master.

Reply:
Well spotted. Have no idea what the epsilon should be but I'll change these to `abs` to start with.

I suspect changing the epsilon should probably go in a separate PR (if you want to make one maybe after we finish this one?).

Reply:
Well spotted. Have no idea what the epsilon should be but I'll change these to `abs` to start with.

I suspect changing the epsilon should probably go in a separate PR (if you want to make one maybe after we finish this one?).

Reply:
Ok have changed this to `Math::abs`.

Reply:
Ok have changed this to `Math::abs`.